### PR TITLE
APP-37: Simple Sign Auth-Screen

### PR DESCRIPTION
**Pull request by hedvigoscar**
Deferred for now:

- Keyboard animation. I love it, but it's too hard to get done right now
- Split the thing out to a module. Given that we did not merge the instant app
  code, I'm left with too little inspiration to get this done in time

### Checklist

- [X] Functionality is covered by an integration test
- [X] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/zignsec/ui/IdentityInputFragment.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                    input.setHelperText(R.string.simple_sign_login_text_field_helper_text)
+                    inputText.setMaxLength(11)
+                    signIn.setText(R.string.simple_sign_sign_in)
+                }
+                Market.DK -> {
+                    input.setHint(R.string.simple_sign_login_text_field_label_dk)
+                    input.setHelperText(R.string.simple_sign_login_text_field_helper_text_dk)
+                    inputText.setMaxLength(10)
+                    signIn.setText(R.string.simple_sign_sign_in_dk)
+                }
```
Comment (hugokallstrom) [inline app/src/main/java/com/hedvig/app/feature/zignsec/ui/IdentityInputFragment.kt:51]: I think we could be very defensive here and show a generic error, or maybe a kind of market unavailable-error. I understand this "shouldn't" happen, but I also think its scary to leave this as a no-op.

**Review by hugokallstrom — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hedvig/app/feature/zignsec/SimpleSignAuthenticationViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            else -> false
+        }
+    }
+
+    private val _isSubmitting = MutableLiveData(false)
+    val isSubmitting: LiveData<Boolean> = _isSubmitting
+
+    private val _zignSecUrl = MutableLiveData<String>()
+    val zignSecUrl: LiveData<String> = _zignSecUrl
+
```
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/zignsec/SimpleSignAuthenticationViewModel.kt:42]: It's used in `ZignSecWebViewFragment` 🙂

File: app/src/main/java/com/hedvig/app/feature/zignsec/SimpleSignAuthenticationViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        _events.value = Event.Error
+    }
+
+    fun startZignSec() {
+        if (isSubmitting.value == true) {
+            return
+        }
+        _isSubmitting.value = true
+        when (data.market) {
+            Market.NO -> {
```
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/zignsec/SimpleSignAuthenticationViewModel.kt:92]: I'm not sure that I'm understanding you entirely here, perhaps you can explain it later in a call 😬

**Comment by hedvigoscar** [comment]
I'm going to merge this for now as we want to be able to test this, but let's look at the last remaining issue when you're back.

**Pull request merged**